pkg/apis/build/v1beta1/source.go

```diff
@@ -59,6 +59,13 @@ type Git struct {
 	//
 	// +optional
 	CloneSecret *string `json:"cloneSecret,omitempty"`
+
+	// ShallowCloneDepth specifies the depth of the shallow clone.
+	// If not specified or set to 0, a full clone will be performed.
+	// Values greater than 0 will create a shallow clone with the specified depth.
+	//
+	// +optional
```

Comment:
I would default this value to 1 so that we do not introduce a breaking change, and add a validation that this must be greater than or equal to 0.

```suggestion
	// +optional
	// +default=1
	// +kubebuilder:validation:Minimum=0
```